src/components/button/button.ts

```diff
@@ -0,0 +1,25 @@
+import RootComponent from '@common/rootComponent';
+import { IRootComponent } from '@common/types';
+import { fromEvent } from 'rxjs';
+
+export interface ButtonHandlers {
+    onClick?: (target: Event) => void;
+}
+
+class Button extends RootComponent {
+    constructor(props: IRootComponent, handlers: ButtonHandlers) {
+        super(props);
+
+        if (this.node) {
+            this.setEventListeners(handlers);
+        }
+    }
+
+    setEventListeners(handlers: ButtonHandlers) {
+        if (handlers.onClick) {
+            fromEvent([this.node], 'click').subscribe(handlers.onClick);
```

Comment:
В случае, когда событие применяется только к одному элементу, указывать массив необязательно
```suggestion
            fromEvent(this.node, 'click').subscribe(handlers.onClick);
```